saraho1123/whats-cookin
# Feature/fav recipestocook functionality

## What is the change?
New design features were added to html and css to add functionality to filter and search methods. Now user can filter and search for certain type of recipes on the main page. Users can also click on a recipe which takes him to a single recipe page where they can find recipe information with directions, ingredients, and the total cost. All three views (all recipes, single recipe, and pantry) were improved where certain buttons are showing up depending on a page you are on.

All comments were removed. Classes, tests files and script.js were refactored and cleaned. Some unused methids were left for the future development.

## What does it fix?
Readability, organization, and DRY.

## Is this a fix or a feature?
feature/small fixes

## Where should the reviewer start?
ingredients.js: 
897 - cleaning

test-data.js 
83 - cleaning

basket.js 
8 - cleaning

Ingredient.js
8 - no changes

Pantry.js 
1 - cleaning code from comments

Recipe.js 
1 - cleaning
12 - naming convention

User.js 
6 - 50 - two new methods were moved from script to User class: `addRecipesToCook(buttons`) & `addFavoriteRecipes(buttons)`. Two new properties were added to constructor

index.html 
10 - end - cleaning the code and removing the default values. Two new buttons were added "Submit Filter" and "Submit Search". Price tag was added to the single recipe page.

styles.css 
27 - new buttons and styles were added

script.js
1 - 29 - new global variables and const for data files were added
34-42 - event listeners for window and multiple html elements were created
43-47 - new variable is created for storing certain values to display on a page
58 - end - new methods were added to allow users to search and filter through the recipes using tags, ingredients, or favorite/recipesToCook filters; a method that takes users to the single recipe page with instructions, ingredients, and the cost; a method that displays pantry of the user to show all available ingredients; a method that initiates the User on load and calls helper functions to display the functional website.

## How should this be tested?
When user open the website, a random user would be generated. On the main page users are able to see the list of all available recipes which they can choose as favorite or to cook in the future. On top of the page they can use filter and search box with available options to sort through the recipes. By clicking on pantry the user will be sent to Pantry page where they can see their ingredients. By clicking on a recipes on all recipes page the user will be sent to a single recipe page with all recipe information.
